packages/prosess-vedtak/src/components/VedtakForm.jsx

```diff
@@ -1,438 +1,388 @@
-  harOverstyrtMedIngenBrev,
-  harBareFritekstbrev,
+  kanKunVelge,
+  harMellomlagretFritekstbrev,
+  harMellomLagretMedIngenBrev,
+  kanHindreUtsending,
+  kanHaAutomatiskVedtaksbrev,
+  filterInformasjonsbehov,
+  harPotensieltFlereInformasjonsbehov,
+  harMellomlagretRedusertUtbetalingArsak,
 } from '@fpsak-frontend/utils/src/formidlingUtils';
-import vedtakBeregningsresultatPropType from '../propTypes/vedtakBeregningsresultatPropType';
+import vedtaksbrevtype from '@fpsak-frontend/kodeverk/src/vedtaksbrevtype';
 import vedtakVilkarPropType from '../propTypes/vedtakVilkarPropType';
 import VedtakInnvilgetPanel from './VedtakInnvilgetPanel';
 import VedtakAvslagPanel from './VedtakAvslagPanel';
 import VedtakAksjonspunktPanel from './VedtakAksjonspunktPanel';
 import styles from './vedtakForm.less';
-import VedtakOverstyrendeKnapp from './VedtakOverstyrendeKnapp';
 import BrevPanel from './brev/BrevPanel';
 import UstrukturerteDokumenter from './UstrukturerteDokumenter';
+import RevurderingPaneler from './revurdering/RevurderingPaneler';
+import redusertUtbetalingArsak from '../kodeverk/redusertUtbetalingArsak';
+import VedtakRevurderingSubmitPanel from './revurdering/VedtakRevurderingSubmitPanel';
+import VedtakSubmit from './VedtakSubmit';
+import vedtakVarselPropType from '../propTypes/vedtakVarselPropType';
 
 const isVedtakSubmission = true;
 
-const kanSendesTilGodkjenning = behandlingStatusKode =>
-  behandlingStatusKode === behandlingStatusCode.BEHANDLING_UTREDES;
+const fieldnames = {
+  SKAL_BRUKE_OVERSTYRENDE_FRITEKST_BREV: 'skalBrukeOverstyrendeFritekstBrev',
+  SKAL_HINDRE_UTSENDING_AV_BREV: 'skalHindreUtsendingAvBrev',
+  OVERSKRIFT: 'overskrift',
+  BRØDTEKST: 'brødtekst',
+  OVERSTYRT_MOTTAKER: 'overstyrtMottaker',
+  BEGRUNNELSE: 'begrunnelse',
+};
 
-const formName = 'VedtakForm';
+const transformRedusertUtbetalingÅrsaker = formikValues =>
+  Object.values(redusertUtbetalingArsak).filter(name =>
+    Object.keys(formikValues).some(key => key === name && formikValues[key]),
+  );
 
-export class VedtakForm extends Component {
-  constructor(props) {
-    super(props);
-    this.onToggleOverstyring = this.onToggleOverstyring.bind(this);
-    this.state = {
-      skalBrukeOverstyrendeFritekstBrev: props.skalBrukeOverstyrendeFritekstBrev,
-    };
-  }
+export const VedtakForm = ({
+  intl,
+  readOnly,
+  behandlingStatus,
+  behandlingresultat,
+  aksjonspunkter,
+  behandlingPaaVent,
+  vedtakVarsel,
+  previewCallback,
+  sprakkode,
+  ytelseTypeKode,
+  alleKodeverk,
+  personopplysninger,
+  arbeidsgiverOpplysningerPerId,
+  tilbakekrevingvalg,
+  vilkar,
+  tilgjengeligeVedtaksbrev,
+  informasjonsbehovVedtaksbrev,
+  dokumentdata,
+  submitCallback,
+  fritekstdokumenter,
+  lagreDokumentdata,
+  overlappendeYtelser,
+  resultatstruktur,
+  simuleringResultat,
+  resultatstrukturOriginalBehandling,
+  bgPeriodeMedAvslagsårsak,
+  medlemskapFom,
+  erRevurdering,
+  behandlingArsaker,
+}) => {
```

Comment:
Et par props her, ja 😅

Reply:
hehe, ja. Kunne sikkert vært forenklet. Men føler det er mer oversiktlig nå som man slipper å forholde seg til initialValues fra redux-form og egen komponent for revurdering